### PR TITLE
WIP: remove unneeded calls to daemon-reload

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
WIP. Remove unneeded calls to daemon-reload. [Testing if CI passes for the moment.]

### CODE REVIEW

**Comment by chipaca** [comment]
Two things:

* please run the unit tests locally before pushing, as otherwise you're needlessly spinning up 50+ machines.
* a lot of integration tests failed with “Unit file [...] changed on disk, 'systemctl --system daemon-reload' recommended.” (afaict all on 14.04 but it might be because we didn't get to see the whole log)

File: systemd/systemd.go [hunk 1]
```diff
@@ -154,6 +163,30 @@ func (*systemd) DaemonReload() error {
 	return err
 }
 
+func (s *systemd) DaemonReloadIfNeeded(serviceNames ...string) error {
+	needReload := false
+	for _, service := range serviceNames {
+		// A status error will happen if the service does not exist anymore.
+		// We do nothing in that case, as in some cases that is the expected
+		// state of the service when this function is called.
```
Comment (niemeyer) [inline systemd/systemd.go]: I realize this is still WIP, but as a heads up the fact we are doing that sounds like a problem on itself, if that's indeed the case. We shouldn't try to reload a daemon that doesn't exist, and we should hopefully have metadata to tell whether that's the case at the call site.

**Comment by niemeyer** [comment]
Marking as blocked as it's WIP.

**Comment by niemeyer** [comment]
Alfonso, I'm closing this as it's WIP and untouched for a month. Please feel free to reopen whenever we can act on it again.

**Pull request closed without merging**